src/lib/hooks/useFavorites.tsx

```diff
@@ -1,19 +1,38 @@
 "use client";
 
-import {useBoolean, useLocalStorage} from "usehooks-ts";
-import {useCallback, useEffect} from "react";
+import { useCallback } from "react";
+import { useLocalStorage } from "usehooks-ts";
 
-const useFavorites = (): [boolean, (_value: boolean) => void] => {
-  const {value: isIntl, setValue: setIsIntl} =  useBoolean(false)
-  const [userIntl, setUserIntl] = useLocalStorage<boolean | undefined>("intl", undefined, {initializeWithValue: false})
+export type Favorite = {
+  uuid: string;
+  title: string;
+  path: string; 
+  units: number;
+};
 
-  const onEvent = useCallback(({coords}: { coords: GeolocationCoordinates }) => {
-    const {latitude, longitude} = coords;
-    setIsIntl(latitude < 24.5 || longitude > -52 || longitude < -170)
-  }, [setIsIntl]);
+const useFavorites = (): {
+  favs: Favorite[];
+  addFav: (_uuid: string, _title: string, _path: string, _units: number) => void;
+  removeFav: (_uuid: string) => void;
+} => {
+  const [favs, setFavs] = useLocalStorage<Favorite[]>("favorites", [], {initializeWithValue: false});
 
-  useEffect(() => navigator.geolocation.getCurrentPosition(onEvent), [onEvent]);
-  return [userIntl !== undefined ? userIntl : isIntl, setUserIntl];
-}
+  const addFav = useCallback(
+    (uuid: string, title: string, path: string, units: number) => {
+      setFavs([...favs, { uuid, title, path, units }]);
+    },
+    [favs, setFavs]
+  );
 
-export default useFavorites;
\ No newline at end of file
+  const removeFav = useCallback(
+    (uuid: string) => {
+      const updatedFavs = favs.filter((fav) => fav.uuid !== uuid);
+      setFavs(updatedFavs);
+    },
+    [favs, setFavs]
+  );
+
```

Comment:
What about a `toggleFav` function? then you can just call that from the components instead of checking if it is a favorite first?